app/services/remove_status_service.rb

```diff
@@ -57,14 +53,37 @@ def remove_reblogs
     end
   end
 
-  def unpush(type, receiver, status)
-    if status.reblog? && !redis.zscore(FeedManager.instance.key(type, receiver.id), status.reblog_of_id).nil?
-      redis.zadd(FeedManager.instance.key(type, receiver.id), status.reblog_of_id, status.reblog_of_id)
+  def unpush_bulk(type, receiver_ids, status)
```

Comment:
I merged the lambdas and the last `redis.piplined` block as you suggested.
However, I'm still not convinced the idea to divide the method to two. They would be `unpush_reblog_bulk` and `unpush_original_bulk` or something sort of. It is a clearly mistake to call `unpush_reblog_bulk` without checking `status.reblog?` and therefore `if status.reblog?` and the content of those two methods should be in a method.